docs/Introduction/Plutarch Terms/Plutarch Constants.md

```diff
@@ -97,6 +97,13 @@ These types deserve some explaination.
 
 That is, if we ask `justTerm` what it will return when evaluated, it responds, "You should interpret the value I give you as either `Nothing` or `Just Integer`." Of course, we know that the result will always be `Just 3`; but this is the general mechanism to declare a function requiring a `Maybe`.
 
+If you don't want to pretend to not know `x` during compile time, another example may be:
+
+```hs 
+hPJustPInteger :: Term s PInteger -> Term s (PMaybe PInteger)
+hPJustPInteger x = pcon (PJust x)
```

Comment:
what's up with the naming? looks good otherwise

Reply:
Following the scheme outlined in the beginning of the docs of naming Plutarch stuff p*, and Haskell stuff h*. It's not merely about strict adherence to those conventions but about detecting misunderstanding. If that question does not make much sense please disregard, it will fall into place later.